### ISSUE TITLE
cannot have a mutable reference to an immutable object

### ISSUE THREAD

**Issue by sameh-farouk** [crystaluniverse/crystallib#76]
after recent v update, I'm getting this error when I try to use crystallib library

```v
error: `freeflowuniverse.crystallib.process.processmap` is immutable, cannot have a mutable reference to an immutable object
   44 | //then you can get it everywhere
   45 | pub fn processmap_get() ?&ProcessMap {
   46 |     mut pm := process.processmap
      |                       ~~~~~~~~~~
   47 |     pm.scan()?
   48 |     return pm
checker summary: 1 V errors, 0 V warnings, 0 V notices
```

I will push a fix on the development_refactor branch

**Comment by medvednikov** [comment]
You can do `mut pm := unsafe { process.processmap }`

**Comment by sameh-farouk** [comment]
fixed in e60662f42729c3c0b18751f744762ea28d0393d6
